bin/minifollowups/pycbc_injection_minifollowup

```diff
@@ -14,20 +14,99 @@
+import logging
+import copy
+import numpy
+import h5py
 import pycbc.workflow as wf
+import pycbc.workflow.minifollowups as mini
+import pycbc.version
+import pycbc.pnutils
 from pycbc.types import MultiDetOptionAction
 from pycbc.events import select_segments_by_definer, coinc
 from pycbc.results import layout
 from pycbc.detector import Detector
-import pycbc.workflow.minifollowups as mini
 from pycbc.workflow.core import resolve_url_to_file
-import pycbc.version, pycbc.pnutils
 from pycbc.io.hdf import SingleDetTriggers, HFile
+from pycbc.conversions import mchirp_from_mass1_mass2
+
 
-parser = argparse.ArgumentParser(description=__doc__[1:])
+def sort_injections(args, inj_group, missed, sort_by):
+    """Return an array of indices to sort the missed injections according to a
+    metric of choice.
+
+    Parameters
+    ----------
+    args : object
+        CLI arguments parsed by argparse.
+    inj_group : h5py group object
+        HDF5 group object containing the injection definition.
+    missed : array
+        Array of indices of missed injections into `inj_group`.
+    sort_by : str
+        Which metric to sort by.
+
+    Returns
+    -------
+    missed_sorted : array
+        Array of indices of missed injections sorted as requested.
+    """
+    if 'optimal_snr' in sort_by:
+        optimal_snrs = [
+            inj_group[dsn][:][missed] for dsn in inj_group.keys()
+            if dsn.startswith('optimal_snr_')
+        ]
+        assert optimal_snrs, 'These injections do not have optimal SNRs'
+
+    if sort_by == 'decisive_optimal_snr':
+        # descending order of decisive (2nd largest) optimal SNR
+        dec_snr = numpy.array([
+            sorted(snrs)[-2] for snrs in zip(*optimal_snrs)
+        ])
+        if args.maximum_decisive_snr is not None:
+            # By setting to 0, these injections will not be considered
+            dec_snr[dec_snr > args.maximum_decisive_snr] = 0
+        sorter = dec_snr.argsort()[::-1]
+        return missed[sorter]
+
+    if sort_by == 'network_optimal_snr':
+        # descending order of network optimal SNR
+        optimal_snrs = numpy.vstack(optimal_snrs)
+        net_opt_snrs_squared = (optimal_snrs ** 2).sum(axis=0)
+        sorter = net_opt_snrs_squared.argsort()[::-1]
+        return missed[sorter]
+
+    if sort_by == 'decisive_eff_dist':
+        # ascending order of decisive (2nd smallest) chirp distance
+        eff_dists = []
+        for ifo in args.single_detector_triggers:
+            eff_dist = Detector(ifo).effective_distance(
+                inj_group['distance'][:][missed],
+                inj_group['ra'][:][missed],
+                inj_group['dec'][:][missed],
+                inj_group['polarization'][:][missed],
+                inj_group['tc'][:][missed],
+                inj_group['inclination'][:][missed]
+            )
+            eff_dists.append(eff_dist)
+        dec_eff_dist = sorted(eff_dists)[-2]
+        mchirp = mchirp_from_mass1_mass2(
```

Comment:
As the eff chirp dist sorting option is mainly provided for legacy / backwards compatibility, how about only importing the conversions and pnutils functions inside this block?

Reply:
Ok